packages/account/src/account.ts

```diff
@@ -542,10 +543,13 @@ export class Account extends AbstractAccount {
 
     const findAssetInput = (assetId: string) =>
       txRequestClone.inputs.find((input) => {
-        if ('assetId' in input) {
+        if (input.type === InputType.Coin) {
           return input.assetId === assetId;
         }
-        if ('recipient' in input) {
+
+        // We only consider the message input if it has no data.
+        // Messages with `data` cannot fund the gas of a transaction.
+        if (input.type === InputType.Message && bn(input.data).isZero()) {
```

Comment:
Adding a test covering this would be great, and then we're good to go.

> cc @Torres-ssf @petertonysmith94

Reply:
@Torres-ssf I assume as you've self assigned you'll add the test for this?